src/dmd/backend/outbuf.d

```diff
@@ -152,34 +254,89 @@ struct Outbuffer
     /**
      * Writes a 32 bit int.
      */
-    void write32(int v);
+    void write32(int v)
+    {
+        if (pend - p < 4)
+            reserve(4);
+        *cast(int *)p = v;
+        p += 4;
+    }
 
     /**
      * Writes a 64 bit long.
      */
-    void write64(long v);
+    version (X86_64)
+        private enum write64Mangle = "_ZN9Outbuffer7write64Ex";
+    else
+        private enum write64Mangle = "_ZN9Outbuffer7write64El";
```

Comment:
I wasn't sure what happened here. Why doesn't `long` mangle to `long long` on x86_64?
The spec says it would do so: https://dlang.org/spec/cpp_interface.html#data-type-compatibility :/